js/background/background.js

```diff
@@ -93,173 +137,150 @@ class AugmentedSteamApi extends Api {
+                                        } else throw new Error("Couldn't retrieve information from URL fragment '" + hashFragment + "'");
+                                    } else throw new Error("Failed to verify state parameter from URL fragment");
+                                } else throw new Error("URL " + url + " doesn't contain a fragment");
+                            } else throw new Error("Couldn't retrieve access token for ITAD authorization");
+                        });
+                } else throw new Error("Denied accessing identity API, can't authorize app");
+            });
+        });
     }
 
-    static async earlyAccessAppIds() {
-        return AugmentedSteamApi._earlyAccessAppIds();    
-    }
+    static isExpired() {
+        let lsEntry = LocalStorage.get("access_token");
+        if (!lsEntry) return true;
 
-    static async dlcInfo({ 'params': params, }) {
-        return AugmentedSteamApi.getEndpoint("v01/dlcinfo", params).then(result => result.data);
+        if (lsEntry.expiry <= window.timestamp()) {
+            LocalStorage.remove("access_token");
+            return true;
+        }
+        ITAD_Api.accessToken = lsEntry.token;
+        ITAD_Api.sync();
+        return false;
     }
 
-    static async expireStorePageData({ 'params': params, }) {
-        CacheStorage.remove(`app_${params.appid}`);
+    static endpointFactoryCached(endpoint, objectStore, multiple, oneDimensional, resultFn) {
+        return async (params, dbKey) => {
+            return super.endpointFactoryCached(endpoint, objectStore, multiple, oneDimensional, resultFn)(Object.assign(params || {}, { access_token: ITAD_Api.accessToken }), dbKey)
+        }
     }
-}
-AugmentedSteamApi.origin = Config.ApiServerHost;
-AugmentedSteamApi._progressingRequests = new Map();
-AugmentedSteamApi._earlyAccessAppIds_promise = null;
 
+    static async sync() {
+        let [ownedApps, ownedPackages, wishlistedApps] = await IndexedDB.get("dynamicStore", ["ownedApps", "ownedPackages", "wishlisted"]);
+        let [lastOwnedApps, lastOwnedPackages, lastWishlistedApps] = await IndexedDB.get("lastItadSync", ["lastOwnedApps", "lastOwnedPackages", "lastWishlisted"]);
 
-class SteamStore extends Api {
-    // static origin = "https://store.steampowered.com/";
-    // static params = { 'credentials': 'include', };
-    // static _progressingRequests = new Map();
-    
-    static async appDetails({ 'params': params, }) {
-        return SteamStore.getEndpoint("/api/appdetails/", params);
-    }
+        let baseJSON = {
+            "version": "02",
+            "data": [],
+        }
 
-    static async appUserDetails({ 'params': params, }) {
-        return SteamStore.getEndpoint("/api/appuserdetails/", params);
-    }
+        function removeDuplicates(from, other) {
+            if (!from) return [];
```

Comment:
For consistency reasons, I suggest you add brackets here and the one below.

Reply:
I'll leave this as is right now, we will do this once #431 has been resolved